src/Illuminate/Collections/Arr.php

```diff
@@ -526,8 +527,8 @@ public static function random($array, $number = null)
 
         $results = [];
 
-        foreach ((array) $keys as $key) {
-            $results[] = $array[$key];
+        foreach ((array) $keys as $index => $key) {
+            $results[$preserveKeys ? $key : $index] = $array[$key];
```

Comment:
This is actually going to run quite slow at runtime, since the CPU has to branch over and over. Better to check the preserve keys is set, and copy the forloop code for both cases.